pkg/statistics/handle/autoanalyze/autoanalyze.go

```diff
@@ -76,6 +79,167 @@ func (sa *statsAnalyze) DeleteAnalyzeJobs(updateTime time.Time) error {
 	})
 }
 
+// CleanupCorruptedAnalyzeJobsOnCurrentInstance cleans up the potentially corrupted analyze job.
+// It only cleans up the jobs that are associated with the current instance.
+func (sa *statsAnalyze) CleanupCorruptedAnalyzeJobsOnCurrentInstance(currentRunningProcessIDs map[uint64]struct{}) error {
+	return statsutil.CallWithSCtx(sa.statsHandle.SPool(), func(sctx sessionctx.Context) error {
+		return CleanupCorruptedAnalyzeJobsOnCurrentInstance(sctx, currentRunningProcessIDs)
+	})
+}
+
+// CleanupCorruptedAnalyzeJobsOnDeadInstances removes analyze jobs that may have been corrupted.
+// Specifically, it removes jobs associated with instances that no longer exist in the cluster.
+func (sa *statsAnalyze) CleanupCorruptedAnalyzeJobsOnDeadInstances() error {
+	return statsutil.CallWithSCtx(sa.statsHandle.SPool(), func(sctx sessionctx.Context) error {
+		return CleanupCorruptedAnalyzeJobsOnDeadInstances(sctx)
+	})
+}
+
+// SelectAnalyzeJobsOnCurrentInstanceSQL is the SQL to select the analyze jobs whose
+// state is `pending` or `running` and the update time is more than 10 minutes ago
+// and the instance is current instance.
+const SelectAnalyzeJobsOnCurrentInstanceSQL = `SELECT id, process_id
+		FROM mysql.analyze_jobs
+		WHERE instance = %?
+		AND state IN ('pending', 'running')
+		AND TIMESTAMPDIFF(MINUTE, update_time, NOW()) > 10`
```

Comment:
Be aware that writing the expression like this will make this SQL unable to make use of the index on `update_time`.
You need to write it something like `update_time > xxx`